### PR TITLE
Add airgap chart install and install docs

### PR DESCRIPTION
**Pull request by meldafrawi**
longhorn/longhorn#812

### CODE REVIEW

File: docs/longhorn-airgap.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Longhorn Airgap install
+
+1. Get Longhorn Deployment manifest file
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply docs/longhorn-airgap.md]: fixed

File: docs/longhorn-airgap.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+`wget https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml`
+
+2. Create Longhorn namespace
+
+  `kubectl create namespace longhorn-system`
+
+
+3. If private registry require authentication, Create `docker-registery` secret in `longhorn-system` namespace:
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply docs/longhorn-airgap.md]: fixed

File: docs/longhorn-airgap.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  * Add your secret name  `SECRET_NAME` to `imagePullSecrets.name` in the following resources
+    * `longhorn-driver-deployer` Deployment
+    * `longhorn-manager` DaemonSet
+    * `longhorn-ui` Deployment
+
+4. Apply the following modifications to Longhorn Deployment manifest file
+
+  * Modify k8s CSI driver components env variables in `longhorn-driver-deployer` Deployment point to your private registry images
+     * CSI_ATTACHER_IMAGE
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply docs/longhorn-airgap.md]: added more details about how to replace variables on longhorn-driver deployment

File: docs/longhorn-airgap.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+     * CSI_PROVISIONER_IMAGE
+     * CSI_NODE_DRIVER_REGISTRAR_IMAGE
+     * CSI_RESIZER_IMAGE
+
+  * Modify Longhorn images to point to your private registry images
+    * longhornio/longhorn-manager
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply docs/longhorn-airgap.md]: added more details about how to replace Longhorn images with private registry images.

File: docs/longhorn-airgap.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  `kubectl -n longhorn-system create secret docker-registry SECRET_NAME --docker-server=REGISTRY_URL --docker-username=REGISTRY_USER --docker-password=REGISTRY_PASSWORD`
+
+
+  * Add your secret name to `longhorn-default-setting` ConfigMap: 
+
+    `registry-secret:  SECRET_NAME `
+
+  * Add your secret name  `SECRET_NAME` to `imagePullSecrets.name` in the following resources
+    * `longhorn-driver-deployer` Deployment
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply docs/longhorn-airgap.md]: added more details about how to add imagePullSecret to manifest file.

File: chart/questions.yml [hunk 6]
```diff
@@ -4,6 +4,54 @@ labels:
...
 namespace: longhorn-system
 questions:
+- variable: image.longhorn.manager
+  default: longhornio/longhorn-manager
+  description: "Specify Longhorn Manager image."
+  type: string
+  label: Longhorn Manager Image
+  group: "Longhorn Images Settings"
+- variable: image.longhorn.managerTag
+  default: master
```
Comment (meldafrawi) [inline chart/questions.yml]: v0.8.0

File: chart/questions.yml [hunk 7]
```diff
@@ -4,6 +4,54 @@ labels:
   io.rancher.certified: experimental
 namespace: longhorn-system
 questions:
+- variable: image.longhorn.manager
+  default: longhornio/longhorn-manager
+  description: "Specify Longhorn Manager image."
```
Comment (meldafrawi) [inline chart/questions.yml]: image name

File: chart/questions.yml [hunk 8]
```diff
@@ -4,6 +4,54 @@ labels:
...
+  default: master
+  description: "Specify Longhorn Manager image tag."
+  type: string
+  label: Longhorn Manager Image tag
+  group: "Longhorn Images Settings"
+- variable: image.longhorn.engine
+  default: longhornio/longhorn-engine
+  description: "Specify Longhorn Engine image."
+  type: string
+  label: Longhorn Manager Image
```
Comment (meldafrawi) [inline chart/questions.yml]: Engine

File: chart/questions.yml [hunk 9]
```diff
@@ -144,6 +197,30 @@ questions:
   group: "Longhorn Default Settings"
   type: string
   default: ""
+- variable: defaultSettings.registrySecret.secretName
+  label: Private registry secret
+  description: "The Kubernetes Secret name"
+  group: "Longhorn Default Settings"
+  type: string
+  default: ""
+- variable: defaultSettings.registrySecret.registryUrl
```
Comment (meldafrawi) [inline chart/questions.yml]: separate from default settings

File: chart/templates/deployment-driver.yaml [hunk 10]
```diff
@@ -58,12 +58,20 @@ spec:
           - name: CSI_DRIVER_REGISTRAR_IMAGE
             value: {{ .Values.csi.driverRegistrarImage }}
           {{- end }}
-          {{- if .Values.csi.attacherReplicaCount }}
+          {{- if .Values.csi.resizerImage }}
+          - name: CSI_RESIZER_IMAGE
+            value: {{ .Values.csi.resizerImage }}
+          {{- end }}
+ {{- if .Values.csi.attacherReplicaCount }}
```
Comment (meldafrawi) [inline chart/templates/deployment-driver.yaml]: indentation

File: docs/longhorn-airgap.md [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+      `helm install longhorn ./chart --namespace longhorn-system`
+
+
+## Using Rancher App
+
+ * In `Longhorn Images Settings` section specify
+
+    * Longhorn Manager Image Name     e.g. `<REGISTRY_URL>/longhorn-manager`
+    * Longhorn Manager Image Tag 
+    * Longhorn Engine Image Name  image name    e.g. <REGISTRY_URL>/longhorn-engine
```
Comment (yasker) [inline docs/longhorn-airgap.md]: Extra `image name`, and missing quote for the image part. Same below.

File: docs/longhorn-airgap.md [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+ * In `Longhorn Images Settings` section specify
+
+    * Longhorn Manager Image Name     e.g. `<REGISTRY_URL>/longhorn-manager`
+    * Longhorn Manager Image Tag 
+    * Longhorn Engine Image Name    e.g. `<REGISTRY_URL>/longhorn-engine`
+    * Longhorn Engine Image Tag 
+    * Longhorn UI Image Name    e.g. `<REGISTRY_URL>/longhorn-instance-manager`
+    * Longhorn UI Image Tag 
+    * Longhorn Instance Manager Image Name    e.g. `<REGISTRY_URL>/longhorn-ui:<LONGHORN_UI_IMAGE_TAG>`
+    * Longhorn Manager Image Tag
```
Comment (yasker) [inline docs/longhorn-airgap.md]: Extra line.

File: docs/longhorn-airgap.md [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+    * Longhorn Engine Image Tag 
+    * Longhorn UI Image Name    e.g. `<REGISTRY_URL>/longhorn-instance-manager`
+    * Longhorn UI Image Tag 
+    * Longhorn Instance Manager Image Name    e.g. `<REGISTRY_URL>/longhorn-ui:<LONGHORN_UI_IMAGE_TAG>`
+    * Longhorn Manager Image Tag 
+
+
+  * In `Longhorn CSI Driver Setting` section specify
+
+    * Longhorn CSI Attacher Image    e.g.  <REGISTRY_URL>/csi-attacher:<CSI_ATTACHER_IMAGE_TAG>
```
Comment (yasker) [inline docs/longhorn-airgap.md]: Quote here too.

File: docs/longhorn-airgap.md (hunk 12 above)
Comment (yasker) [reply docs/longhorn-airgap.md]: `    * Longhorn Manager Image Tag ` is still here. 

Also the previous names and URLs are wrong.

File: docs/longhorn-airgap.md [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+## Using Rancher App
+
+  * In `Longhorn Images Settings` section specify
+    * Longhorn Manager Image Name     e.g. `<REGISTRY_URL>/longhorn-manager`
+    * Longhorn Manager Image Tag 
+    * Longhorn Engine Image Name    e.g. `<REGISTRY_URL>/longhorn-engine`
+    * Longhorn Engine Image Tag 
+    * Longhorn UI Image Name    e.g. `<REGISTRY_URL>/longhorn-ui` 
+    * Longhorn UI Image Tag 
+    * Longhorn Instance Manager Image Name    e.g.  `<REGISTRY_URL>/longhorn-instance-manager`
```
Comment (yasker) [inline docs/longhorn-airgap.md]: Should be `Longhorn Instance Manager Image`

Comment (meldafrawi) [reply docs/longhorn-airgap.md]: fixed

**Pull request merged**